execute/plugin.go

```diff
@@ -60,9 +60,13 @@ type Plugin struct {
 	estimateProvider      cciptypes.EstimateProvider
 	lggr                  logger.Logger
 
-	observationOptimizer optimizers.ObservationOptimizer
 	// state
 	contractsInitialized bool
+	// this cache remembers commit root details to optimize DB lookups.
+	commitRootsCache cache.CommitsRootsCache
+
+	// TODO: remove this, it can be transient. Also its logger is never initialized.
+	observationOptimizer optimizers.ObservationOptimizer
```

Comment:
Can't we remove it as a part of this PR? If not, could please create a ticket? I think we easily forget about these TODOs in the code